tests/python/pants_test/engine/legacy/test_query_integration.py

```diff
@@ -0,0 +1,32 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest, ensure_daemon
+
+
+class ListQueryIntegrationTest(PantsRunIntegrationTest):
+  """Test the --query option, which is intended to subsume multiple other options and goals.
+
+  If v2 rules are made to satisfy both existing options such as --owner-of and their equivalent in
+  the --query option, the testing for that functionality should be moved into this file (or into
+  unit testing), and the separate option or goal should eventually be deprecated.
+  """
+  # TODO(#7346): test changes-since or changes-for-diffspec queries! It's currently difficult to set
```

Comment:
Can take a look at the harness in https://github.com/pantsbuild/pants/blob/master/tests/python/pants_test/engine/legacy/test_changed_integration.py for an example.